teamcity: failed tests on master: testrace/TestRemoveRangeWithoutGC

The following tests appear to have failed:

[#46303](https://teamcity.cockroachdb.com/viewLog.html?buildId=46303):

```
--- FAIL: testrace/TestRemoveRangeWithoutGC (0.007s)
Race detected!
I161111 21:01:00.193434 21960 storage/engine/rocksdb.go:340  opening in memory rocksdb instance
I161111 21:01:00.194893 21960 gossip/gossip.go:237  [n?] initial resolvers: []
W161111 21:01:00.194992 21960 gossip/gossip.go:1055  [n?] no resolvers found; use --join to specify a connected node
I161111 21:01:00.195136 21960 base/node_id.go:62  NodeID set to 1
I161111 21:01:00.208331 21960 storage/store.go:1187  [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I161111 21:01:00.208534 21960 gossip/gossip.go:280  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:47054" > attrs:<> locality:<> 
I161111 21:01:00.209115 22005 storage/replica_proposal.go:328  [s1,r1/1:/M{in-ax}] new range lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 900.000124ms following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0s [physicalTime=1970-01-01 00:00:00.000000123 +0000 UTC]
I161111 21:01:00.211816 21960 storage/engine/rocksdb.go:340  opening in memory rocksdb instance
I161111 21:01:00.212991 21960 gossip/gossip.go:237  [n?] initial resolvers: [127.0.0.1:47054]
W161111 21:01:00.213094 21960 gossip/gossip.go:1057  [n?] no incoming or outgoing connections
I161111 21:01:00.213248 21960 base/node_id.go:62  NodeID set to 2
I161111 21:01:00.219295 21884 gossip/client.go:125  [n2] started gossip client to 127.0.0.1:47054
I161111 21:01:00.221133 21960 storage/store.go:1187  [n2,s2]: failed initial metrics computation: [n2,s2]: system config not yet available
I161111 21:01:00.221344 21960 gossip/gossip.go:280  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:42165" > attrs:<> locality:<> 
I161111 21:01:00.228620 21960 storage/replica_raftstorage.go:445  [s1,r1/1:/M{in-ax}] generated snapshot 4c0cc78d at index 13 in 129.505��s.
I161111 21:01:00.230411 21960 storage/store.go:3126  [s1,r1/1:/M{in-ax}] streamed snapshot: kv pairs: 32, log entries: 3
I161111 21:01:00.231226 21981 storage/replica_raftstorage.go:589  [s2,r1/?:{-}] applying preemptive snapshot at index 13 (id=4c0cc78d, encoded size=16, 1 rocksdb batches, 3 log entries)
I161111 21:01:00.232284 21981 storage/replica_raftstorage.go:592  [s2,r1/?:/M{in-ax}] applied preemptive snapshot in 0.001s
I161111 21:01:00.234454 21960 storage/replica_command.go:3245  [s1,r1/1:/M{in-ax}] change replicas: read existing descriptor range_id:1 start_key:"" end_key:"\377\377" replicas:<node_id:1 store_id:1 replica_id:1 > next_replica_id:2 
I161111 21:01:00.238295 21984 storage/replica.go:2055  [s1,r1/1:/M{in-ax}] proposing ADD_REPLICA {NodeID:2 StoreID:2 ReplicaID:2}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:2 StoreID:2 ReplicaID:2}]
I161111 21:01:00.370378 22149 storage/raft_transport.go:432  raft transport stream to node 1 established
I161111 21:01:00.391393 21960 storage/replica_command.go:3245  [s1,r1/1:/M{in-ax}] change replicas: read existing descriptor range_id:1 start_key:"" end_key:"\377\377" replicas:<node_id:1 store_id:1 replica_id:1 > replicas:<node_id:2 store_id:2 replica_id:2 > next_replica_id:3 
I161111 21:01:00.401009 22166 storage/replica.go:2055  [s1,r1/1:/M{in-ax}] proposing REMOVE_REPLICA {NodeID:1 StoreID:1 ReplicaID:1}: [{NodeID:2 StoreID:2 ReplicaID:2}]
E161111 21:01:00.404821 22011 storage/replica_proposal.go:520  [s1,r1/1:/M{in-ax}] unable to add to replica GC queue: queue disabled
I161111 21:01:00.405984 21960 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161111 21:01:00.406086 21960 util/stop/stopper.go:468  quiescing; tasks left:
1      storage/replica_proposal.go:636
1      kv/txn_coord_sender.go:918
I161111 21:01:00.406343 22167 storage/replica_proposal.go:631  [s1,r1/1:/M{in-ax}] unable to gossip first range; hasLease=false, err=node unavailable; try another peer
I161111 21:01:00.407029 21960 util/stop/stopper.go:468  quiescing; tasks left:
1      storage/replica_proposal.go:636
==================
WARNING: DATA RACE
Read at 0x00c420092370 by goroutine 381:
  github.com/cockroachdb/cockroach/pkg/storage_test.(*multiTestContext).RangeLookup()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_test.go:579 +0x82
  github.com/cockroachdb/cockroach/pkg/kv.(*rangeDescriptorCache).performRangeLookup()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_cache.go:443 +0x559
  github.com/cockroachdb/cockroach/pkg/kv.(*rangeDescriptorCache).lookupRangeDescriptorInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_cache.go:327 +0x649
  github.com/cockroachdb/cockroach/pkg/kv.(*rangeDescriptorCache).LookupRangeDescriptor()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_cache.go:266 +0xb3
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).getDescriptor()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:401 +0xc3
  github.com/cockroachdb/cockroach/pkg/kv.(*RangeIterator).Seek()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_iter.go:132 +0x3f5
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).divideAndSendBatchToRanges()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:670 +0x248
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:576 +0x32e
  github.com/cockroachdb/cockroach/pkg/kv.(*TxnCoordSender).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/txn_coord_sender.go:416 +0x7f1
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).sendInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:585 +0x16e
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).send()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:709 +0x63c
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).(github.com/cockroachdb/cockroach/pkg/internal/client.send)-fm()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:311 +0x7c
  github.com/cockroachdb/cockroach/pkg/internal/client.sendAndFill()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:416 +0x1ab
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:311 +0xfd
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).updateLiveness.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:307 +0x54e
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:466 +0x46
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Exec()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:518 +0x233
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:467 +0x297
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).updateLiveness()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:308 +0x158
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).heartbeat()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:188 +0x4ed
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).StartHeartbeat.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:140 +0x3b4
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x8a

Previous write at 0x00c420092370 by goroutine 177:
  github.com/cockroachdb/cockroach/pkg/storage_test.(*multiTestContext).populateDB()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_test.go:603 +0x37d
  github.com/cockroachdb/cockroach/pkg/storage_test.(*multiTestContext).restartStore()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_test.go:827 +0x220
  github.com/cockroachdb/cockroach/pkg/storage_test.TestRemoveRangeWithoutGC()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2558 +0x6cb
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:610 +0xc9

Goroutine 381 (running) created at:
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x73
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).StartHeartbeat()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:152 +0x122
  github.com/cockroachdb/cockroach/pkg/storage_test.(*multiTestContext).addStore()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_test.go:776 +0x178b
  github.com/cockroachdb/cockroach/pkg/storage_test.(*multiTestContext).Start()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_test.go:299 +0xab6
  github.com/cockroachdb/cockroach/pkg/storage_test.TestRemoveRangeWithoutGC()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2523 +0x225
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:610 +0xc9

Goroutine 177 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:646 +0x52f
  testing.RunTests.func1()
      /usr/local/go/src/testing/testing.go:793 +0xb9
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:610 +0xc9
  testing.RunTests()
      /usr/local/go/src/testing/testing.go:799 +0x4ba
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:743 +0x12f
  github.com/cockroachdb/cockroach/pkg/storage_test.TestMain()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/main_test.go:57 +0x286
  main.main()
      github.com/cockroachdb/cockroach/pkg/storage/_test/_testmain.go:752 +0x1b5
==================
==================
WARNING: DATA RACE
Read at 0x00c4206be3b0 by goroutine 381:
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).RangeLookup()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:245 +0xf3
  github.com/cockroachdb/cockroach/pkg/storage_test.(*multiTestContext).RangeLookup()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_test.go:579 +0x107
  github.com/cockroachdb/cockroach/pkg/kv.(*rangeDescriptorCache).performRangeLookup()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_cache.go:443 +0x559
  github.com/cockroachdb/cockroach/pkg/kv.(*rangeDescriptorCache).lookupRangeDescriptorInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_cache.go:327 +0x649
  github.com/cockroachdb/cockroach/pkg/kv.(*rangeDescriptorCache).LookupRangeDescriptor()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_cache.go:266 +0xb3
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).getDescriptor()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:401 +0xc3
  github.com/cockroachdb/cockroach/pkg/kv.(*RangeIterator).Seek()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_iter.go:132 +0x3f5
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).divideAndSendBatchToRanges()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:670 +0x248
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:576 +0x32e
  github.com/cockroachdb/cockroach/pkg/kv.(*TxnCoordSender).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/txn_coord_sender.go:416 +0x7f1
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).sendInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:585 +0x16e
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).send()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:709 +0x63c
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).(github.com/cockroachdb/cockroach/pkg/internal/client.send)-fm()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:311 +0x7c
  github.com/cockroachdb/cockroach/pkg/internal/client.sendAndFill()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:416 +0x1ab
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:311 +0xfd
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).updateLiveness.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:307 +0x54e
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:466 +0x46
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Exec()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:518 +0x233
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:467 +0x297
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).updateLiveness()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:308 +0x158
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).heartbeat()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:188 +0x4ed
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).StartHeartbeat.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:140 +0x3b4
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x8a

Previous write at 0x00c4206be3b0 by goroutine 177:
  github.com/cockroachdb/cockroach/pkg/kv.NewDistSender()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:145 +0x74
  github.com/cockroachdb/cockroach/pkg/storage_test.(*multiTestContext).populateDB()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_test.go:608 +0x316
  github.com/cockroachdb/cockroach/pkg/storage_test.(*multiTestContext).restartStore()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_test.go:827 +0x220
  github.com/cockroachdb/cockroach/pkg/storage_test.TestRemoveRangeWithoutGC()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2558 +0x6cb
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:610 +0xc9

Goroutine 381 (running) created at:
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x73
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).StartHeartbeat()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:152 +0x122
  github.com/cockroachdb/cockroach/pkg/storage_test.(*multiTestContext).addStore()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_test.go:776 +0x178b
  github.com/cockroachdb/cockroach/pkg/storage_test.(*multiTestContext).Start()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_test.go:299 +0xab6
  github.com/cockroachdb/cockroach/pkg/storage_test.TestRemoveRangeWithoutGC()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2523 +0x225
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:610 +0xc9

Goroutine 177 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:646 +0x52f
  testing.RunTests.func1()
      /usr/local/go/src/testing/testing.go:793 +0xb9
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:610 +0xc9
  testing.RunTests()
      /usr/local/go/src/testing/testing.go:799 +0x4ba
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:743 +0x12f
  github.com/cockroachdb/cockroach/pkg/storage_test.TestMain()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/main_test.go:57 +0x286
  main.main()
      github.com/cockroachdb/cockroach/pkg/storage/_test/_testmain.go:752 +0x1b5
==================
==================
WARNING: DATA RACE
Read at 0x00c4206be3a0 by goroutine 381:
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).RangeLookup()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:248 +0x1b9
  github.com/cockroachdb/cockroach/pkg/storage_test.(*multiTestContext).RangeLookup()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_test.go:579 +0x107
  github.com/cockroachdb/cockroach/pkg/kv.(*rangeDescriptorCache).performRangeLookup()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_cache.go:443 +0x559
  github.com/cockroachdb/cockroach/pkg/kv.(*rangeDescriptorCache).lookupRangeDescriptorInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_cache.go:327 +0x649
  github.com/cockroachdb/cockroach/pkg/kv.(*rangeDescriptorCache).LookupRangeDescriptor()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_cache.go:266 +0xb3
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).getDescriptor()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:401 +0xc3
  github.com/cockroachdb/cockroach/pkg/kv.(*RangeIterator).Seek()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_iter.go:132 +0x3f5
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).divideAndSendBatchToRanges()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:670 +0x248
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:576 +0x32e
  github.com/cockroachdb/cockroach/pkg/kv.(*TxnCoordSender).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/txn_coord_sender.go:416 +0x7f1
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).sendInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:585 +0x16e
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).send()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:709 +0x63c
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).(github.com/cockroachdb/cockroach/pkg/internal/client.send)-fm()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:311 +0x7c
  github.com/cockroachdb/cockroach/pkg/internal/client.sendAndFill()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:416 +0x1ab
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:311 +0xfd
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).updateLiveness.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:307 +0x54e
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:466 +0x46
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Exec()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:518 +0x233
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:467 +0x297
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).updateLiveness()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:308 +0x158
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).heartbeat()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:188 +0x4ed
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).StartHeartbeat.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:140 +0x3b4
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x8a

Previous write at 0x00c4206be3a0 by goroutine 177:
  github.com/cockroachdb/cockroach/pkg/kv.NewDistSender()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:145 +0x74
  github.com/cockroachdb/cockroach/pkg/storage_test.(*multiTestContext).populateDB()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_test.go:608 +0x316
  github.com/cockroachdb/cockroach/pkg/storage_test.(*multiTestContext).restartStore()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_test.go:827 +0x220
  github.com/cockroachdb/cockroach/pkg/storage_test.TestRemoveRangeWithoutGC()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2558 +0x6cb
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:610 +0xc9

Goroutine 381 (running) created at:
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x73
  github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).StartHeartbeat()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:152 +0x122
  github.com/cockroachdb/cockroach/pkg/storage_test.(*multiTestContext).addStore()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_test.go:776 +0x178b
  github.com/cockroachdb/cockroach/pkg/storage_test.(*multiTestContext).Start()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_test.go:299 +0xab6
  github.com/cockroachdb/cockroach/pkg/storage_test.TestRemoveRangeWithoutGC()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2523 +0x225
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:610 +0xc9

Goroutine 177 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:646 +0x52f
  testing.RunTests.func1()
      /usr/local/go/src/testing/testing.go:793 +0xb9
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:610 +0xc9
  testing.RunTests()
      /usr/local/go/src/testing/testing.go:799 +0x4ba
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:743 +0x12f
  github.com/cockroachdb/cockroach/pkg/storage_test.TestMain()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/main_test.go:57 +0x286
  main.main()
      github.com/cockroachdb/cockroach/pkg/storage/_test/_testmain.go:752 +0x1b5
==================
E161111 21:01:00.412045 21960 storage/store.go:1050  [n1,s1,r1/?:/M{in-ax}]: unable to add replica to GC queue: queue disabled
W161111 21:01:00.415214 21995 storage/replica.go:3804  [s1] could not acquire lease for range gossip: range 1 was not found
W161111 21:01:00.415300 21995 storage/store.go:1236  [s1] error gossiping system config: range 1 was not found
W161111 21:01:00.415535 21994 storage/replica.go:3804  [s1] could not acquire lease for range gossip: range 1 was not found
W161111 21:01:00.415624 21994 storage/store.go:1236  [s1] error gossiping first range descriptor: range 1 was not found
I161111 21:01:00.721758 22078 storage/replica_proposal.go:328  [s2,r1/2:/M{in-ax}] new range lease replica {2 2 2} 1970-01-01 00:00:00.900000124 +0000 UTC 240h0m1.800000003s following replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 900.000124ms [physicalTime=1970-01-11 00:00:01.800000126 +0000 UTC]
I161111 21:01:00.726472 22139 util/stop/stopper.go:468  quiescing; tasks left:
1      storage/client_test.go:501
I161111 21:01:00.726747 22094 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
E161111 21:01:00.727106 22115 storage/node_liveness.go:141  [hb] failed liveness heartbeat: node unavailable; try another peer
I161111 21:01:00.728337 22094 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
E161111 21:01:00.729919 22115 storage/node_liveness.go:141  [hb] failed liveness heartbeat: node unavailable; try another peer
I161111 21:01:00.730083 22094 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161111 21:01:00.731482 22059 http2_server.go:276  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:42165->127.0.0.1:38689: use of closed network connection
I161111 21:01:00.731578 22061 http2_client.go:1053  transport: http2Client.notifyError got notified that the client transport was broken EOF.
I161111 21:01:00.732115 22094 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161111 21:01:00.732385 22094 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/storage/client_test.go
+++ b/pkg/storage/client_test.go
@@ -568,17 +568,6 @@ func (m *multiTestContext) FirstRange() (*roachpb.RangeDescriptor, error) {
 	return descs[len(descs)-1], nil
 }
 
-// RangeLookup implements the RangeDescriptorDB interface. It looks up the
-// descriptors for the given (meta) key.
-func (m *multiTestContext) RangeLookup(
-	ctx context.Context, key roachpb.RKey, desc *roachpb.RangeDescriptor, useReverseScan bool,
-) ([]roachpb.RangeDescriptor, []roachpb.RangeDescriptor, *roachpb.Error) {
-	// DistSender's RangeLookup function will work correctly, as long as
-	// multiTestContext's FirstRange() method returns the correct descriptor for the
-	// first range.
-	return m.distSenders[0].RangeLookup(ctx, key, desc, useReverseScan)
-}
-
 func (m *multiTestContext) makeStoreConfig(i int) storage.StoreConfig {
 	var cfg storage.StoreConfig
 	if m.storeConfig != nil {
@@ -597,14 +586,30 @@ func (m *multiTestContext) makeStoreConfig(i int) storage.StoreConfig {
 	return cfg
 }
 
+var _ kv.RangeDescriptorDB = mtcRangeDescriptorDB{}
+
+type mtcRangeDescriptorDB struct {
+	*multiTestContext
+	ds **kv.DistSender
+}
+
+func (mrdb mtcRangeDescriptorDB) RangeLookup(
+	ctx context.Context, key roachpb.RKey, desc *roachpb.RangeDescriptor, useReverseScan bool,
+) ([]roachpb.RangeDescriptor, []roachpb.RangeDescriptor, *roachpb.Error) {
+	return (*mrdb.ds).RangeLookup(ctx, key, desc, useReverseScan)
+}
+
 func (m *multiTestContext) populateDB(idx int, stopper *stop.Stopper) {
 	retryOpts := base.DefaultRetryOptions()
 	retryOpts.Closer = stopper.ShouldQuiesce()
 	m.distSenders[idx] = kv.NewDistSender(kv.DistSenderConfig{
-		Clock:             m.clock,
-		RangeDescriptorDB: m,
-		TransportFactory:  m.kvTransportFactory,
-		RPCRetryOptions:   &retryOpts,
+		Clock: m.clock,
+		RangeDescriptorDB: mtcRangeDescriptorDB{
+			multiTestContext: m,
+			ds:               &m.distSenders[idx],
+		},
+		TransportFactory: m.kvTransportFactory,
+		RPCRetryOptions:  &retryOpts,
 	}, m.gossips[idx])
 	ambient := log.AmbientContext{Tracer: tracing.NewTracer()}
 	sender := kv.NewTxnCoordSender(

